src/omnisharp/engines/IEngine.ts

```diff
@@ -8,22 +8,36 @@ import { LaunchTarget } from '../launcher';
 import { LaunchInfo } from '../OmnisharpManager';
 import { Options } from '../options';
 import Disposable from '../../Disposable';
+import { OmniSharpServer } from '../server';
+import { LanguageMiddlewareFeature } from '../LanguageMiddlewareFeature';
+import OptionProvider from '../../observers/OptionProvider';
+import { Advisor } from '../../features/diagnosticsProvider';
+import TestManager from '../../features/dotnetTest';
+import { EventStream } from '../../EventStream';
 
 export interface IEngine {
     start(
         cwd: string,
         args: string[],
         launchTarget: LaunchTarget,
         launchInfo: LaunchInfo,
-        options: Options
+        options: Options,
     ): Promise<void>;
     stop(): Promise<void>;
+    registerProviders(
```

Comment:
Allow the engines to register VS Code providers as needed. LSP won't necessarily need as many.